src/bone.js

```diff
@@ -1313,6 +1313,9 @@ class Bone {
       opts.uniqueKeys = opts.uniqueKeys.map((field) => this.unalias(field));
     }
 
+    // records might change when filter through custom setters
+    records = instances.map(instance => instance.getRaw());
```

Comment:
line 1326 will never be reached, should be removed

Reply:
indeed, removed now.